lti_consumer/tests/unit/test_lti_xblock.py

```diff
@@ -1329,6 +1329,40 @@ def test_launch_callback_endpoint_deep_linking(self):
         # Check response
         self.assertEqual(response.status_code, 200)
 
+    def test_launch_callback_endpoint_deep_linking_by_student(self):
+        """
+        Test that the callback endpoint errors out if students try to do a deep link launch.
+        """
+        self.xblock.runtime.get_user_role.return_value = 'student'
+        mock_user_service = Mock()
+        mock_user_service.get_external_user_id.return_value = 2
+        self.xblock.runtime.service.return_value = mock_user_service
+
+        self.xblock.course.display_name_with_default = 'course_display_name'
+        self.xblock.course.display_org_with_default = 'course_display_org'
+
+        # Enable deep linkin
+        self.xblock.lti_advantage_deep_linking_enabled = True
+
+        # Get LTI client_id
+        client_id = get_lti_1p3_launch_info(block=self.xblock)['client_id']
+
+        # Craft request sent back by LTI tool
+        request = make_request('', 'GET')
+        request.query_string = (
+            "client_id={}&".format(client_id) +
+            "redirect_uri=http://tool.example/launch&" +
+            "state=state_test_123&" +
+            "nonce=nonce&" +
+            "login_hint=oidchint&" +
+            "lti_message_hint=deep_linking_launch"
+        )
+
+        response = self.xblock.lti_1p3_launch_callback(request)
+
+        # Check response
+        self.assertEqual(response.status_code, 400)
```

Comment:
Should we use ``403`` here instead? As we need a specific role for users to do an LTI Launch, ``403`` seems more appropriate to me. What do you think?

Reply:
@shimulch This is a student facing error page, not sure we should throw/display a permission denied error for a student.
Students will only hit this error if they are trying to bypass the `staff` requirement and make a Deep Link Launch without permissions.